### PR TITLE
Arc - document intercepted self-invocation as a non-standard feature

### LINKED ISSUE #41545: Documentation: Clarify possible self-invocation as CDI extension
### Description

It would be appreciated if some person of authority on the subject (@Ladicek maybe?) could clarify, and maybe even add a section about this to https://quarkus.io/guides/cdi-reference that in Quarkus, beans are indeed intercepted on self-invocation, even though this is not possible with standard CDI. For example, the following code works in Quarkus:
```java
@ApplicationScoped
@Transactional
public class MyService {
    @Transactional(REQUIRES_NEW)
    protected void doStuffInNewTransaction() {
        System.out.println("do stuff, but in a new transaction");
    }

    @Scheduled(every = "PT1S")
    void someSchedule() {
        System.out.println("do stuff");
        // This call goes through a transactional interceptor! (TransactionalInterceptorRequiresNew),
        // even though in pure CDI self-invocation does not perform interception.
        doStuffInNewTransaction();
    }
}
```

However, in CDI this would not be possible and instead none of the annotations on `doStuffInNewTransaction` would have any effect for the above invocation. See also [CDI-414](https://issues.redhat.com/browse/CDI-414)

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by manovotn**
Fixes #41545

### CODE REVIEW

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 1]
```diff
@@ -1069,6 +1069,35 @@ public class NoopAsyncObserverExceptionHandler implements AsyncObserverException
 }
 ----
 
+=== Intercepted self-invocation
+
+Quarkus supports what is known as intercepted self-invocation or just self-interception - a scenario, where CDI bean invokes its own intercepted method from within another method while triggering any associated interceptors.
```
Comment (Ladicek) [inline docs/src/main/asciidoc/cdi-reference.adoc:1074]:
```suggestion
Quarkus supports what is known as intercepted self-invocation or just self-interception - a scenario where CDI bean invokes its own intercepted method from within another method while triggering any associated interceptors.
```

**Review by Ladicek — APPROVED** [review]
The text seems a bit hasty, but I honestly cannot figure out a better one myself, so I'll just stay silent :-)

**Comment by manovotn** [comment]
> The text seems a bit hasty, but I honestly cannot figure out a better one myself, so I'll just stay silent :-)

Well, I didn't think of any better way to phrase it either. I can give it some more thought.

**Review by mkouba — APPROVED** [review]

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 2]
```diff
@@ -1069,6 +1069,34 @@ public class NoopAsyncObserverExceptionHandler implements AsyncObserverException
...
+  void doSomethingElse() {
+    doSomething();<2>
+  }
+
+}
+----
+<1> One or more interceptor bindings; `@Transactional` is just an example.
+<2> Non-intercepted method invoking another method from the same bean that has associated binding(s); this will trigger interception.
+
+In the above example, any code executed inside the `doSomething()` method triggers interception - in this case becomes transactional - regardless of whether the invocation originated directly from `MyService#doSomething` or `MyService#doSomethingElse`.
```
Comment (Ladicek) [inline docs/src/main/asciidoc/cdi-reference.adoc:1098]:
```suggestion
In the above example, any code calling the `doSomething()` method triggers interception - in this case, the method becomes transactional.
This is regardless of whether the invocation originated directly from the `MyService` bean (such as `MyService#doSomethingElse`) or from some other bean.
```

**Comment by Ladicek** [comment]
LGTM otherwise.

**Review by Ladicek — APPROVED** [review]

**Pull request merged**